### PR TITLE
Repair coronal and saggital contour extraction

### LINKED ISSUE #340: repair indices_to_pos() function
it returns bad values.
https://github.com/pytrip/pytripgui/issues/84

### LINKED ISSUE #342: dicom2trip: sanity check
add a simple check to avoid weird errors when slice_thickness is larger than slice distance.
(had that case with an internal dataset)

### PR DESCRIPTION
**Pull request by nbassler**
issues
- [x] contours in sagittal and coronal are messed up due to missing sorting
- [x] check offsets in TRiP: is this still properly handled?
- [x] how to handle overlapping slices (i.e. thickness = 5 mm and distance 3.5 mm)

closes #342 
closes #340

### CODE REVIEW

**Comment by nbassler** [comment]
how to handle wider slices: rely on slice positions, not slice thickness, after discussing locally.

File: pytrip/cube.py [hunk 1]
```diff
@@ -394,7 +394,7 @@ def write_trip_header(self, path):
             output_str += "slice_no  position  thickness  gantry_tilt\n"
             for i, item in enumerate(self.slice_pos):
                 output_str += "{:3d}{:16.4f}{:13.4f}{:14.4f}\n".format(i + 1, item,
-                                                                       self.slice_distance, 0)  # 0 gantry tilt
+                                                                       self.slice_thickness, 0)  # 0 gantry tilt
```
Comment (nbassler) [inline pytrip/cube.py]: I made to hed+ctx files with either slice_distance or slice_thickness using the overlapping slice DICOM-CT i have. Made a proton plan on both of them, and TRiP mapped it correctly in both cases. So I prefer passing slice_thickness here as it is more correct, and slice_distance can be derived anyway from the z_position.

File: pytrip/vdx.py [hunk 2]
```diff
@@ -788,6 +789,17 @@ def create_dicom_contour_data(self, i):
...
+    def _sort_slices(self):
+        """ Sorts all slices stored in self for increasing z.
+        This is needed for displaying Saggital and Coronal view.
+        """
+        # slice_in_frame is only given by VDX, and these are also the only frames which need to be sorted
+        # it seems. DICOM apparently have proper structure already. Nonetheless, this function is also
+        # applied to DICOM contours.
+
+        if hasattr(self.slices[0], "slice_in_frame"):
+            self.slices = sorted(self.slices, key=lambda _slice: _slice.slice_in_frame, reverse=True)
```
Comment (grzanka) [inline pytrip/vdx.py]: Better to use `sort` method of `list` class which is sorting in place:::

```
self.slices.sort(key=lambda _slice: _slice.slice_in_frame, reverse=True)
```

**Comment by grzanka** [comment]
@nbassler Feel free to squash merge when the tests are done. I may have limited internet connectivity this evening. Tomorrow morning I can make new releases

**Pull request merged**